python/ray/dag/compiled_dag_node.py

```diff
@@ -1036,6 +1055,7 @@ def _get_or_compile(
                     self,
                     reader_and_node_list,
                     typ=type_hint,
+                    num_shm_buffers=self._max_buffered_inputs,
```

Comment:
Does the num buffers only apply for the DAG input or is it to all channels in the DAG?

Reply:
it is applied to all channels. This way, there's a additional benefit that the last stage is not blocked when ray.get is not called.